assets/bigquery/encounter-events.sql.j2

```diff
@@ -105,7 +123,46 @@ WITH
     ON
       encountered_vessel_id = encountered_vessel2.vessel_id
       AND (EXTRACT(YEAR FROM encounter.start_time)= encountered_vessel2.year OR encountered_vessel2.year IS NULL)
-  )
+  ),
+  create_event_region_matches as (
+    SELECT encounters.event_id, id, layer,
+    FROM
+      complete_encounter_event AS encounters
+    JOIN gridded_regions r 
+    ON encounters.s2_cell = r.s2_cell
+    WHERE ST_INTERSECTS(encounters.geo, r.geo)
+  ),
+  event_by_layer_by_id as (
+    select 
+      event_id, layer, id
+    from create_event_region_matches
+    group by 1, 2, 3
+  ),
+  events_by_layer as(
+    select
+      event_id, CONCAT('"', layer, '":' ,TO_JSON_STRING(array_agg(id))) as json_frament
```

Comment:
typo in 'json_frament' should be 'json_fragment'.   That typo comes from my original code :-)

Reply:
Fixed